helix-core/src/main/java/org/apache/helix/HelixManagerProperty.java

```diff
@@ -80,27 +86,32 @@ public long getHealthReportLatency() {
     return _healthReportLatency;
   }
 
-  public RealmAwareZkClient.RealmAwareZkConnectionConfig getZkConnectionConfig() {
-    return _zkConnectionConfig;
+  public RealmAwareZkClient.RealmAwareZkConnectionConfig getRealmAwareZkConnectionConfig() {
+    return _realmAwareZkConnectionConfig;
   }
 
   public RealmAwareZkClient.RealmAwareZkClientConfig getZkClientConfig() {
     return _zkClientConfig;
   }
 
+  public HelixZkClient.ZkConnectionConfig getZkConnectionConfig() {
+    return _zkConnectionConfig;
+  }
+
   public static class Builder {
     private String _version;
     private long _healthReportLatency;
     private HelixCloudProperty _helixCloudProperty;
-    private RealmAwareZkClient.RealmAwareZkConnectionConfig _zkConnectionConfig;
+    private RealmAwareZkClient.RealmAwareZkConnectionConfig _realmAwareZkConnectionConfig;
     private RealmAwareZkClient.RealmAwareZkClientConfig _zkClientConfig;
+    private HelixZkClient.ZkConnectionConfig _zkConnectionConfig;
 
     public Builder() {
     }
 
     public HelixManagerProperty build() {
       return new HelixManagerProperty(_version, _healthReportLatency, _helixCloudProperty,
-          _zkConnectionConfig, _zkClientConfig);
+          _realmAwareZkConnectionConfig, _zkClientConfig, _zkConnectionConfig);
```

Comment:
Question: For this public function, shall we keep the original signature and add this as a new one?

Reply:
I agree, we should have 2 distinct methods